src/Bridge/Doctrine/DBAL/Types/TypesDumper.php

```diff
@@ -101,23 +141,26 @@ private function getTypeCode(string $classname, string $enumClass, string $type,
     if (!\class_exists($classname::class)) {
         class $classname extends \\{$baseClass}
         {
-            public const NAME = '$name';
-
-            protected function getEnumClass(): string
-            {
-                return \\{$enumClass}::class;
-            }
-
-            public function getName(): string
-            {
-                return static::NAME;
-            }
+$code
         }
     }
 
 PHP;
     }
 
+    private function valueAsCode($value): string
```

Comment:
I don't think there is a need for this method. Let's just keep the value as passed.

Reply:
But then we'd get an PHP error (undefined constant) in case of a string value. We must at least add quotes to strings, otherwise they end up as toplevel PHP constants in the generated code.

Reply:
I just noticed that I can just use `var_export()`. :facepalm:

Reply:
Ok, got it. I'm fine with the `var_export` for this purpose 👍🏻 

>  int, float and bool are valid enum value types AFAIK

Only int and string are officially valid enum types actually: https://github.com/Elao/PhpEnums/blob/0fbdccfe49ee47edecdfa91980e4e46b8584b8ff/src/EnumInterface.php#L18

Reply:
Got it, thanks. I'll tackle the tests next.